File: 2486-append-characters-to-string-to-make-subsequence/2486-append-characters-to-string-to-make-subsequence.py
class Solution(object):
    def appendCharacters(self, s, t):
        i, j, n, m = 0, 0, len(s), len(t)
        while i < n and j < m:
            if s[i] == t[j]:
                i += 1
                j += 1
            else:
                i += 1
        
        return m - j
        

File: 2486-append-characters-to-string-to-make-subsequence/2486-append-characters-to-string-to-make-subsequence.java
class Solution {
    public int appendCharacters(String s, String t) {
        int i = 0, j = 0, n = s.length(), m = t.length();
        while(i < n && j < m) {
            if(s.charAt(i) == t.charAt(j)) {
                i++; j++;
            }
            else i++;
        }

        return m - j;
    }
}